content/src/content/jcr_root/apps/core/wcm/sandbox/components/carousel/v1/carousel/clientlibs/site/js/carousel.js

```diff
@@ -0,0 +1,212 @@
+                    refreshActive();
+                });
+            }
+
+            var indicators = that._elements["indicator"];
+            if (indicators) {
+                for (var i = 0; i < indicators.length; i++) {
+                    (function(index) {
+                        indicators[i].addEventListener("click", function(event) {
+                            navigate(index);
+                        });
+                    })(i);
+                }
+            }
+        }
+
+        function refreshActive() {
+            var items = that._elements["item"];
+            var indicators = that._elements["indicator"];
+
+            if (items) {
+                if (Array.isArray(items)) {
+                    for (var i = 0; i < items.length; i++) {
+                        if (i === parseInt(that._active)) {
+                            items[i].classList.add("cmp-carousel__item--active");
+                            items[i].removeAttribute("aria-hidden");
+                            indicators[i].classList.add("cmp-carousel__indicator--active");
+                            indicators[i].setAttribute("aria-selected", true);
+                        } else {
+                            items[i].classList.remove("cmp-carousel__item--active");
+                            items[i].setAttribute("aria-hidden", true);
+                            indicators[i].classList.remove("cmp-carousel__indicator--active");
+                            indicators[i].setAttribute("aria-selected", false);
+                        }
+                    }
+                } else {
+                    // only one item
+                    items.classList.add("cmp-carousel__item--active");
+                    indicators.classList.add("cmp-carousel__indicator--active");
+                }
+            }
+        }
+
+        function navigate(index) {
+            that._active = index;
+            refreshActive();
+        }
+    }
+
+    function readData(element) {
+        var data = element.dataset;
+        var options = [];
+        var capitalized = IS;
+        capitalized = capitalized.charAt(0).toUpperCase() + capitalized.slice(1);
+        var reserved = ["is", "hook" + capitalized];
+
+        for (var key in data) {
+            if (data.hasOwnProperty(key)) {
+                var value = data[key];
+
+                if (key.indexOf(NS) === 0) {
+                    key = key.slice(NS.length);
+                    key = key.charAt(0).toLowerCase() + key.substring(1);
+
+                    if (reserved.indexOf(key) === -1) {
+                        options[key] = value;
+                    }
+                }
+            }
+        }
+
+        return options;
+    }
+
+    function onDocumentReady() {
+        var elements = document.querySelectorAll(selectors.self);
+        for (var i = 0; i < elements.length; i++) {
+            new Carousel({ element: elements[i], options: readData(elements[i]) });
```

Comment:
Is `options` needed? I couldn't find a use of it in the Carousel class.

Reply:
I would rather leave it there for now, we may want to add some options in the next iteration (e.g. slide duration etc.)